app/components/blogFilter.tsx

```diff
@@ -0,0 +1,83 @@
+'use client'; // Tento řádek je potřeba, aby byla komponenta považována za klientskou
+
+import React, { useState } from 'react';
+import Link from "next/link";
+import Image from "next/image";
+import {metaData} from "../config";
+
+export function BlogFilterComponent({ allBlogs }: { allBlogs: any[] }) {
+    const [selectedTag, setSelectedTag] = useState('all'); // Stav pro výběr tagu
+    const [filteredBlogs, setFilteredBlogs] = useState(allBlogs); // Počáteční stav pro všechny blogy
+
+    // Funkce pro filtrování blogů podle tagu
+    const handleTagSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        const selected = e.target.value;
+        setSelectedTag(selected);
+
+        // Filtrování blogů
+        const filtered = selected === 'all'
+            ? allBlogs
+            : allBlogs.filter((post) => post.metadata.tags.includes(selected));
+
+        setFilteredBlogs(filtered); // Nastavíme filtrované blogy
+    };
+
+    return (
+        <div>
+            <select onChange={handleTagSelect} value={selectedTag}>
+                <option value="all">Všechny</option>
+                <option value="jóga">Jóga</option>
+                <option value="koučink">Koučink</option>
+            </select>
+
+            <div className="flex flex-wrap">
+
+                {/* Zobrazení filtrovaných blogů */}
+                {filteredBlogs
+                    .sort((a, b) => {
```

Comment:
I'd consider move this sorting logic to it's function and declare it above this component, or better yet move it to new utils file